File: questions/1663. Smallest String With A Given Numeric Value/smallest.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
        string s(n, 'a');
        k -= n;
        int i = n - 1;
        while(k > 0){
            int tmp = min(k, 25);
            s[i] += tmp;
            k -= tmp;
            i--;
        }
        return s;
    }
};

File: questions/1663. Smallest String With A Given Numeric Value/smallest.java
class Solution {
    public String getSmallestString(int n, int k) {
        StringBuilder str = new StringBuilder();
        for(int i = 0; i < n; i++){
            str.append('a');
        }
        k -= n;
        int i = n - 1;
        while(k > 0){
            int tmp = Math.min(k, 25);
            str.setCharAt(i, (char)((int)str.charAt(i) + tmp));
            k -= tmp;
            i--;
        }
        return str.toString();
    }
}
